### PR TITLE
Refactor timer.

### PR DESCRIPTION
**Pull request by Beomjun**
Refactor timer.c and timer.h.

### CODE REVIEW

**Comment by kesl** [comment]
Can one of the admins verify this patch?

**Comment by Beomjun** [comment]
ok to test

**Comment by joohyun0115** [comment]
ok to test

**Comment by kesl** [comment]
CI : Test PASSed.
Refer to this link for build results: http://keslproject.iptime.org:8082/job/k-hypervisor_pull_request/225/

File: hypervisor/include/timer.h [hunk 1]
```diff
@@ -4,11 +4,14 @@
 #include "hvmm_types.h"
 #include "arch_types.h"
 
-#define MAX_TIMER 8
+#define MAX_TIMERS_COUNT    8
+#define TRUE                1
+#define FALSE               0
+#define ERROR_COUNT         -1
 
 typedef void(*timer_callback_t)(void *pdata);
```
Comment (Wonseok) [inline hypervisor/include/timer.h:12]: What is the pdata?

File: hypervisor/include/timer.h [hunk 2]
```diff
@@ -4,11 +4,14 @@
 #include "hvmm_types.h"
 #include "arch_types.h"
 
-#define MAX_TIMER 8
+#define MAX_TIMERS_COUNT    8
```
Comment (Wonseok) [inline hypervisor/include/timer.h:7]: s/COUNT/NUM or NUMBER/

File: hypervisor/include/timer.h [hunk 3]
```diff
@@ -4,11 +4,14 @@
...
+#define TRUE                1
+#define FALSE               0
+#define ERROR_COUNT         -1
```
Comment (Wonseok) [inline hypervisor/include/timer.h:10]: What does ERROR_COUNT   mean?

File: hypervisor/include/timer.h [hunk 4]
```diff
@@ -4,11 +4,14 @@
...
 
 typedef void(*timer_callback_t)(void *pdata);
 
-struct timer_val {
+struct timer {
```
Comment (Wonseok) [inline hypervisor/include/timer.h:14]: This structure can represent timer?

File: hypervisor/timer.c [hunk 5]
```diff
@@ -130,38 +126,46 @@ static hvmm_status_t timer_requset_irq(uint32_t irq)
...
+    struct timer_struct new_timer;
+    new_timer.timer.callback = timer->callback;
+    new_timer.timer.interval_us = timer->interval_us;
+    new_timer.count_per_irq = calculate_count_per_irq(timer->interval_us);
+
+    /*
+     * TODO: Storing new_time into array of timers needs a lock mechanism
+     * for preventing concurrent access.
+     */
+    timers[timers_index++] = new_timer;
```
Comment (Wonseok) [inline hypervisor/timer.c:168]: Do not use ++ expression here. Please separate it to two lines.

**Pull request closed without merging**